ext/v8/object.cc

```diff
@@ -98,6 +130,294 @@ namespace rr {
+    Locker lock(object);
+
+    return String::Maybe(object.getIsolate(), object->ObjectProtoToString(Context(r_context)));
+  }
+
+  VALUE Object::GetConstructorName(VALUE self) {
+    Object object(self);
+    Locker lock(object);
+
+    return String(object.getIsolate(), object->GetConstructorName());
+  }
+
+  VALUE Object::InternalFieldCount(VALUE self) {
+    Object object(self);
+    Locker lock(object);
+
+    return INT2FIX(object->InternalFieldCount());
+  }
+
+  VALUE Object::GetInternalField(VALUE self, VALUE index) {
+    Object object(self);
+    Locker lock(object);
+
+    return Value(object.getIsolate(), object->GetInternalField(NUM2INT(index)));
+  }
+
+  VALUE Object::SetInternalField(VALUE self, VALUE index, VALUE value) {
+    Object object(self);
+    Locker lock(object);
+
+    object->SetInternalField(NUM2INT(index), *Value(value));
+
+    return Qnil;
+  }
+
+  VALUE Object::HasOwnProperty(VALUE self, VALUE r_context, VALUE key) {
+    Object object(self);
+    Locker lock(object);
+
+    return Bool::Maybe(object->HasOwnProperty(Context(r_context), *Name(key)));
+  }
+
+  VALUE Object::HasRealNamedProperty(VALUE self, VALUE r_context, VALUE key) {
+    Object object(self);
+    Locker lock(object);
+
+    return Bool::Maybe(object->HasRealNamedProperty(Context(r_context), *Name(key)));
+  }
+
+  VALUE Object::HasRealIndexedProperty(VALUE self, VALUE r_context, VALUE index) {
+    Object object(self);
+    Locker lock(object);
+
+    return Bool::Maybe(object->HasRealIndexedProperty(Context(r_context), NUM2INT(index)));
+  }
+
+  VALUE Object::HasRealNamedCallbackProperty(VALUE self, VALUE r_context, VALUE key) {
+    Object object(self);
+    Locker lock(object);
+
+    return Bool::Maybe(object->HasRealNamedCallbackProperty(Context(r_context), *Name(key)));
+  }
+
+  VALUE Object::GetRealNamedPropertyInPrototypeChain(VALUE self, VALUE r_context, VALUE key) {
+    Object object(self);
+    Locker lock(object);
+
+    return Value::Maybe(object.getIsolate(), object->GetRealNamedPropertyInPrototypeChain(
+      Context(r_context),
+      *Name(key)
+    ));
+  }
+
+  VALUE Object::GetRealNamedProperty(VALUE self, VALUE r_context, VALUE key) {
+    Object object(self);
+    Locker lock(object);
+
+    return Value::Maybe(object.getIsolate(), object->GetRealNamedProperty(
+      Context(r_context),
+      *Name(key)
+    ));
+  }
+
+  VALUE Object::GetRealNamedPropertyAttributes(VALUE self, VALUE r_context, VALUE key) {
+    Object object(self);
+    Locker lock(object);
+
+    return Enum<v8::PropertyAttribute>::Maybe(object->GetRealNamedPropertyAttributes(
+      Context(r_context),
+      *Name(key)
+    ));
+  }
+
+  VALUE Object::GetRealNamedPropertyAttributesInPrototypeChain(VALUE self, VALUE r_context, VALUE key) {
+    Object object(self);
+    Locker lock(object);
+
+    return Enum<v8::PropertyAttribute>::Maybe(object->GetRealNamedPropertyAttributesInPrototypeChain(
+      Context(r_context),
+      *Name(key)
```

Comment:
I'm seeing a bunch of cases where you have to dereference explicitly. Shouldn't this conversion handle it? https://github.com/cowboyd/therubyracer/blob/upgrade-to-v8-4.5/ext/v8/ref.h#L70-L72

If not, perhaps the typedef of `v8::Handle` to `v8::Local` is confusing it? maybe if we change that to a `v8::Local` from `v8::Handle`....